helpers/integration.go

```diff
@@ -180,63 +178,103 @@ func (s SortTaskRunsByStartTime) Less(i int, j int) bool {
 	return s[i].GetStartTime().Before(s[j].GetStartTime())
 }
 
-// CalculateIntegrationPipelineRunOutcome checks the Tekton results for a given PipelineRun and calculates the overall outcome.
+// IntegrationPipelineRunOutcome is struct for pipeline outcome metadata
+type IntegrationPipelineRunOutcome struct {
+	pipelineRunSucceeded bool
+	pipelineRun          *tektonv1beta1.PipelineRun
+	// map: task name to results
+	results map[string]*AppStudioTestResult
+}
+
+// HasPipelineRunSucceeded returns true when pipeline in outcome succeeded
+func (ipro *IntegrationPipelineRunOutcome) HasPipelineRunSucceeded() bool {
+	return ipro.pipelineRunSucceeded
+}
+
+// HasOutcomeSucceeded returns general outcome
 // If any of the tasks with the TEST_OUTPUT result don't have the `result` field set to SUCCESS or SKIPPED, it returns false.
-func CalculateIntegrationPipelineRunOutcome(adapterClient client.Client, ctx context.Context, logger logr.Logger, pipelineRun *tektonv1beta1.PipelineRun) (bool, error) {
-	var results []*AppStudioTestResult
-	var err error
-	// Check if the pipelineRun finished from the condition of status
-	if !HasPipelineRunFinished(pipelineRun) {
-		logger.Info(fmt.Sprintf("PipelineRun %s in namespace %s has not finished", pipelineRun.Name, pipelineRun.Namespace))
-		return false, nil
+func (ipro *IntegrationPipelineRunOutcome) HasOutcomeSucceeded() bool {
```

Comment:
but it also covers status of PLR, not just tasks

Reply:
I think that it's hard to find a suitable name here because `HasPipelineRunSucceeded` checks if the pipelineRun finished successfully without erroring out (in the Tekton sense), and `HasOutcomeSucceeded` checks both if the pipelineRun finished successfully and if the individual tasks in it reported success via the `TEST_OUTPUT`.

Maybe something like `HasPipelineRunFinishedSuccessfully` and `HasPipelineRunPassedTesting`?

Reply:
renamed